### PR TITLE
GEODE-9496: Enhance the usage of newly introduced system property

### PR DESCRIPTION
**Pull request by gesterzhou**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by mhansonp — APPROVED** [review]

**Review by boglesby — APPROVED** [review]

**Review by nabarunnag — APPROVED** [review]

File: geode-core/src/test/java/org/apache/geode/internal/lang/SystemPropertyHelperTest.java [hunk 1]
```diff
@@ -107,6 +107,17 @@ public void getBooleanPropertyReturnsGemfirePrefixIfGeodeMissing() {
     System.clearProperty(gemfirePrefixProperty);
   }
 
+  @Test
+  public void getBooleanPropertyReturnsEnableRetryOnPdxSerializationException() {
+    String testProperty = "enableQueryRetryOnPdxSerializationException";
+    String gemfirePrefixProperty = "gemfire." + testProperty;
```
Comment (jchen21) [inline geode-core/src/test/java/org/apache/geode/internal/lang/SystemPropertyHelperTest.java:113]: Better add additional test with `geode.` prefix as property name.

File: geode-core/src/main/java/org/apache/geode/internal/lang/SystemPropertyHelper.java [hunk 2]
```diff
@@ -71,6 +71,9 @@
 
   public static final String USE_HTTP_SYSTEM_PROPERTY = "useHTTP";
 
+  public static final String ENABLE_QUERY_RETRY_ON_PDX_SERIALIZATION_EXCEPTION =
```
Comment (jchen21) [inline geode-core/src/main/java/org/apache/geode/internal/lang/SystemPropertyHelper.java:74]: Better add some comments for the new system property. Add what is the default value for this property?

**Review by jchen21 — APPROVED** [review]
Is the new system property added to Apache Geode document? Things like what this system property is about, its default value and when it can be turned on/off etc.

**Review by Bill — CHANGES_REQUESTED** [review]
This is a good change. I'll approve as soon as you've added some comments in `SystemPropertyHelper` describing the new property.

**Comment by gesterzhou** [comment]
> This is a good change. I'll approve as soon as you've added some comments in `SystemPropertyHelper` describing the new property.

D

**Review by Bill — APPROVED** [review]
Approved!

**Review by jchen21 — APPROVED** [review]

**Pull request merged**